### ISSUE TITLE
AWSAuthUI "Forgot your username?" Feature 

### ISSUE THREAD

**Issue by BillBunting** [aws-amplify/aws-sdk-ios#2337]
**Which AWS Services is the feature request for?**
AWSMobileClient, AWSAuthUI, AWSCognito

**Is your feature request related to a problem? Please describe.**
Users of an app often forget their cognito usernames.  `AWSMobileClient.default().showSignIn` has a "Forgot your password?" feature, but does not allow a user to recall their username. 

**Describe the solution you'd like**
Add "Forgot your user name" (or similar) in addition to "Forgot your password" (or combine the two as "Forgot your user name or password" to direct to a new combined forgot user name or password view).  

Selecting "Forgot your user name" should prompt for either email or mobile phone number then send by email or text message the registered user's user name.

**Comment by lawmicha** [comment]
hi @BillBunting, this sounds like a neat feature. I will bring this up with Cognito to see this is supported on the service side.

**Comment by wooj2** [comment]
I've submitted a feature request to the Cognito team on the AWS Forums to support a "forgot your user name" workflow.  The request can be found here:
https://forums.aws.amazon.com/thread.jspa?threadID=323959

**Comment by BillBunting** [comment]
This issue is OBE for my app. I modified the AWSUserPoolsSignIn forgot your password view controller storyboard locally to add more descriptive text to the forgot your password feature to help users understand their options, "Enter your registered username, email, or mobile phone to reset your password. Phone number format +1NNNNNNN". For my use case, this is acceptable. The user pool is configured to collect email and phone number, which may be used to sign in or recover a lost account. My app will be moving away from user pools in favor of Sign in with Apple, Facebook, or Google federated sign in. Exiting users may still login with their username and password. Apple recently rejected my app because it collected phone number, against their policy. I resolved the rejection by removing the ability to create a new account with user pools (as my user pool was already collecting phone number as a required field and can not easily be removed).
